pycaret/time_series.py

```diff
@@ -121,6 +122,21 @@ def setup(
         Number of steps ahead to take to evaluate forecast.
 
 
+    sp: int, default = None
```

Comment:
Thinking about this some more, we can have `seasonal_period` accept both int and string values. If it is int, then we simply use that for `sp` internally in `create_model` and `tune_model`. If it is a string, then we use your mapping to get the `sp` value for `create_model` and harmonics of that `sp` value for `tune_model`.

Reply:
got it, then `sp` will take both `int` and `str`. In case of str will mapped to an int value according to the predefined values in `freq_to_sp`.

Reply:
Correct. But I would prefer to maintain a separate name for the argument (such as `seasonal_period` so as to not overwrite the internal `sp` parameter) in case we need to refer to that later for tuning, etc.